specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,456 @@
+
+```rust
+trait SBTNFT {
+  fn nft_total_supply(&self) -> U64
+  // here we index by token id instead of by kind id (as done in `sbt_tokens_by_owner`)
+  fn nft_tokens_for_owner(&self, account_id: AccountId, from_index: Option<U64>, limit: Option<u64>) -> Vec<Token>
+  fn nft_supply_for_owner(&self, account_id: AccountId) -> U64
+```
+
+### Events
+
+```typescript
+type SbtEventKind {
+  standard: "nep393";
+  version: "1.0.0";
+  event: "mint" | "recover" | "renew" | "revoke" | "burn" | "soul_transfer" | "kill";
+  data: Mint | Recover | Renew | Revoke | Burn | SoulTransfer | Kill;
+}
+
+/// An event minted by the Registry when new SBT is created.
+type Mint {
+  ctr: AccountId    // SBT Contract recovering the tokens
+  owner: AccountId, // holder of the newly minted SBT
+  tokens: []u64,    // newly minted token ids
+  memo?: string,    // optional message
+}
+
+
+/// An event emitted when a recovery process succeeded to reassign SBT, usually due to account
+/// access loss. This action is usually requested by the owner, but executed by an issuer,
+/// and doesn't trigger Soul Transfer.
+/// Must be emitted by a SBT registry.
+type Recover {
+  ctr: AccountId         // SBT Contract recovering the tokens
+  old_owner: AccountId;  // current holder of the SBT
+  new_owner: AccountId;  // destination account.
+  tokens: []u64;  // list of token ids.
+  memo?: string;  // optional message
+}
+
+/// An event emitted when a existing tokens are renewed.
+/// Must be emitted by a SBT registry.
+type Renew {
+  ctr: AccountId  // SBT Contract renewing the tokens
+  tokens: []u64;  // list of token ids.
+  memo?: string;  // optional message
+}
+
+/// An event emitted when an existing tokens are revoked.
+/// Revoked tokens should not be listed in a wallet.
+/// Must be emitted by a SBT registry.
+type Revoke {
+  ctr: AccountId  // SBT Contract revoking the tokens
+  tokens: []u64;  // list of token ids.
+  memo?: string;  // optional message
+}
+
+/// An event emitted when an existing tokens are burned.
+/// Must be emitted by a SBT registry.
+type Burn {
+  ctr: AccountId  // SBT Contract revoking the tokens
+  tokens: []u64;  // list of token ids.
+  memo?: string;  // optional message
+}
+
+
+/// An event emitted when soul transfer is happening: all SBTs owned by `from` are transferred
+/// to `to`, and the `from` account is killed (can't receive any new SBT).
+/// Must be emitted by a SBT registry.
+/// Registry MUST emit `Kill` whenever the soul transfer happens.
+type SoulTransfer {
+  from: AccountId;
+  to: AccountId;
+  memo?: string;   // optional message
+}
+
+/// An event emitted when the `account` is killed within the emitting registry.
+/// Must be emitted by a SBT registry.
+/// Registry must add the `account` to a blocklist and prohibit issuing SBTs to this account
+/// in the future
+/// Must be emitted by a SBT registry.
+type Kill {
+  account: AccountId;
+  memo?: string;   // optional message
+}
```

Comment:
Shouldn't be there be a `Born` event, when the account is registered, in contrary to the `Kill` one? If no, what's the point of the event, how can it be used?

Reply:
We have `Mint` which will instrument indexer to update the account status. I don't think we need `Born`. But happy to explore it further if there is a valid use case, which is not covered by existing event.s